### ISSUE TITLE
Inconsistent naming and returns from datasets_create and update_dataset

### ISSUE THREAD

**Issue by toqduj** [SciCatProject/pyscicat#33]
Naming of these methods isn't consistent, the second probably should be dataset_update (and the first dataset_create). Secondly, datasets_create returns a json model, while update_dataset returns a pid. 
Something to tackle for the refactor, methinks.

**Comment by dylanmcreynolds** [comment]
Yes, there has been a lot of talk about a refactor to make more consistent names throughout, especially as we think about adding more functionality to support proposals, samples, etc. @nitrosx and have been talking about this, but haven't had time to implement yet.

**Comment by toqduj** [comment]
I can imagine. If it would help, we could reserve a few days for a micro-code-camp early next year?

**Comment by dylanmcreynolds** [comment]
Yeah, that would be cool. I think so much of our momentum this year will towards getting the new backend out.
